src/main/java/eu/arrowhead/application/skeleton/consumer/ConsumerMain.java

```diff
@@ -0,0 +1,435 @@
+import eu.arrowhead.common.dto.shared.ServiceInterfaceResponseDTO;
+import eu.arrowhead.common.dto.shared.SystemResponseDTO;
+
+@SpringBootApplication
+@ComponentScan(basePackages = { CommonConstants.BASE_PACKAGE, "ai.aitia" })
+public class ConsumerMain implements ApplicationRunner {
+
+	// =================================================================================================
+	// members
+
+	@Autowired
+	private ArrowheadService arrowheadService;
+
+	@Autowired
+	private SSLProperties sslProperties;
+
+	private final Logger logger = LogManager.getLogger(ConsumerMain.class);
+
+	private List<SystemResponseDTO> systems;
+
+	private List<ServiceDefinitionResponseDTO> services;
+
+	private List<ServiceInterfaceResponseDTO> interfaces;
+
+	@Value(CommonConstants.$SERVICEREGISTRY_ADDRESS_WD)
+	private String serviceRegistryAddress;
+
+	@Value(CommonConstants.$SERVICEREGISTRY_PORT_WD)
+	private int serviceRegistryPort;
+
+	// =================================================================================================
+	// methods
+
+	// ------------------------------------------------------------------------------------------------
+	public static void main(final String[] args) {
+		SpringApplication.run(ConsumerMain.class, args);
+	}
+
+	// -------------------------------------------------------------------------------------------------
+	@Override
+	public void run(final ApplicationArguments args) throws Exception {
+
+		if (args.getSourceArgs().length != 1) {
+			logger.error("You must specify exactly one path! (E.g.: ../example.json)");
+			return;
+		}
+
+		final List<AuthRule> newRules = getRules(args.getSourceArgs()[0]);
+
+		if (newRules == null) {
+			logger.error("Reading the file was unsuccessful!");
+			return;
+		}
+
+		systems = getSystems();
+		services = getServices();
+		interfaces = getInterfaces();
+
+		updateAuthRules(newRules);
+	}
+
+	// =================================================================================================
+	// assistant methods
+
+	// -------------------------------------------------------------------------------------------------
+	private List<AuthRule> getRules(final String filename) {
+
+		JsonNode newRulesJson;
+		try {
+			newRulesJson = new ObjectMapper().readTree(new File(filename));
+		} catch (final IOException e) {
+			logger.error(e.getMessage());
+			return null;
+		}
+		final List<AuthRule> rules = new ArrayList<>();
+
+		for (final JsonNode rule : newRulesJson) {
+			try {
+				final AuthRule newRule = Utilities.fromJson(rule.toString(), AuthRule.class);
+				// Setting the defaults if no interfaces are given
+				if (newRule.getInterfaces().size() == 0) {
+					newRule.setInterfaces(new ArrayList<String>(ConsumerConstants.DEFAULT_INTERFACE_NAMES));
+				}
+				rules.add(newRule);
+			} catch (final Exception e) {
+				logger.error("Could not create authorization rule (invalid json format): " + e.getMessage());
```

Comment:
If the json format is incorrect (for example braces or commas are missing), I'm throwing an exception and returning `null` above, as you suggested.
Otherwise, I am creating the `AuthRule `objects one by one. In some cases It migth not violate the json format, but cannot be interpreted as an `AuthRule `object, for example:
```
[
   {
      "consumeeeer":"testconsumer1",
      "provider":"systemtype=test",
      "service":"test-service",
      "interfaces":[
      ]
   },
   {
      "consumer":"testconsumer2",
      "provider":"testprovider",
      "service":"test-service",
      "interfaces":[
      ]
   }
]
```
It this case, there is a typo in the first rule, so it logs out that the first rule cannot be created, but the cycle goes forward and creates the second one.
My intention was that if one rule was in an invalid format, still apply the others. Should I modify the code so no rules will be applied if one of them is incorrect?

Reply:
> My intention was that if one rule was in an invalid format, still apply the others. Should I modify the code so no rules will be applied if one of them is incorrect?

I think yes you should. This is a user input, it can be expected to being completely adequate.